src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobScheduler.scala

```diff
@@ -289,16 +289,18 @@ class JobScheduler @Inject()(val scheduleHorizon: Period,
           val streamForJob = streams.find(_.jobName == job.name)
 
           streamForJob.foreach { stream =>
-            stream.schedule.recurrences.foreach { recurRemaining =>
-              if (recurRemaining == 0) {
-                log.info("Disabling job that reached a zero-recurrence count!")
-
-                val disabledJob = scheduleBasedJob.copy(disabled = true)
-                jobsObserver.apply(JobDisabled(job, """Job '%s' has exhausted all of its recurrences and has been disabled.
-                                                      |Please consider either removing your job, or updating its schedule and re-enabling it.
-                                                    """.stripMargin.format(job.name)))
-                replaceJob(scheduleBasedJob, disabledJob)
-              }
+            stream.schedule match {
+              case Iso8601Schedule(_,_,_,_,_,Some(recurRemaining),_) =>
+                if (recurRemaining == 0) {
```

Comment:
This could be folded into `case`:
```scala
case Iso8601Schedule(_,_,_,_,_,Some(0),_) =>
```
or even
```scala
case iso8601Schedule: Iso8601Schedule if iso8601Schedule.recurrences == Some(0) =>
```
(the latter has the advantage that it references the `recurrences` field by name and the reader does not have to count fields).